Backend/Remora.Discord.Core/IOptional.cs

```diff
@@ -33,6 +34,17 @@ public interface IOptional
         /// <summary>
         /// Gets a value indicating whether the optional contains a value.
         /// </summary>
+        [Obsolete("Prefer IOptional.IsSpecified")]
         bool HasValue { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the optional value is logically present.
+        /// </summary>
+        bool IsSpecified { get; }
```

Comment:
`Optional<T>.HasValue` is named that way for two reasons - one, it *is* checking for the presence of a value: any value, including `null`. `null` is typically used to denote the absence of a value, yes, but it is very much a distinct value in and of itself. Two, it's named the same as `Nullable<T>.HasValue` for consistency and to follow the principle of least surprise.

`Optional<T>` mirrors `Nullable<T>` in a lot of ways, and I see no compelling reason to rename this property.